scripts/travis-run.sh

```diff
@@ -1,6 +1,41 @@
 #!/bin/bash
 set -euo pipefail
 
+# determine recipes to build
+# case 1: env matrix changed or this is a cron job. In this case consider all recipes.
+set +e
+PACKAGES=""
+git diff --exit-code --name-only $TRAVIS_COMMIT_RANGE scripts/env_matrix.yml
+if [ $? -eq 1 ] || [ $TRAVIS_EVENT_TYPE == "cron" ]
+then
+    echo "considering all recipes because either env matrix was changed or build is triggered via cron"
+else
+    # case 2: consider only recipes that (a) changed since the last build on master, or (b) changed in this pull request compared to the target branch.
+    if [ $TRAVIS_PULL_REQUEST == "false" ]
+    then
+        RANGE=$TRAVIS_COMMIT_RANGE
+    else
+        RANGE="$TRAVIS_BRANCH HEAD"
+    fi
+    # obtain recipes that changed in this commit
+    RECIPES=$(git diff --exit-code --relative=recipes --name-only $RANGE recipes/*/meta.yaml recipes/*/*/meta.yaml)
```

Comment:
The motivation is that only changes in the meta.yaml would lead to a new package build. At least, the build number would have to be updated. We might need to revisit this when the new pinning system arrives.